### DISCUSSION TITLE
`getFunctionSelector` for `AbiFunction`

### DISCUSSION THREAD

**Discussion by hexcowboy** [wevm/viem#676] [Idea / Feature Request]
Right now I don't see any utility to get a function selector from an `AbiFunction`. It would be useful to have this to avoid having to convert the `AbiFunction` to a string that can be parsed by the current `getFunctionSelector`, which only accepts ABI function strings right now.

Any work being done on this? I could start working on an implementation and submit a PR.

**Comment by jxom** [comment]
Go for it

**Reply by hexcowboy** [reply]
would you prefer reusing the same `getFunctionSelector` util, changing the parameter to take either `string | AbiFunction`? or would you prefer a new function be made?

**Reply by jxom** [reply]
Same makes sense
